src/App.tsx

```diff
@@ -1,61 +1,136 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState<null | number>(null);
+  const [selectUserError, setSelectUserError] = useState(false);
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const addTodo = (titleString: string, user: number) => {
+    const lastTodoId = Math.max(...visibleTodos.map(todo => todo.id));
```

Comment:
Let's create function for get new todo id and move it to separate file `helpers`